### PR TITLE
Use `Try` node

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
Update pylint to use new `Try` node which replaces the old `TryExcept` and `TryFinally` nodes.

Refs: https://github.com/PyCQA/astroid/pull/1867

### CODE REVIEW

**Comment by cdce8p** [comment]
Thanks for all your work guys! I don't find much time lately to work on pylint, unfortunately but it's great to see it's still going strong. Especially thanks to you @jacobtylerwalls! The astroid improvements look awesome and there's even support for Python 3.12 already. Happy to see that my `Try` node rewrite finally gets completed and merged as well.

File: tests/functional/a/async_functions.py [hunk 1]
```diff
@@ -39,6 +39,8 @@ async def complex_function(this, function, has, more, arguments, than,
         return than
     try:
         return one
+    except TypeError:
```
Comment (Pierre-Sassoulas) [inline tests/functional/a/async_functions.py:42]: What was happening here ?

File: tests/lint/unittest_lint.py [hunk 2]
```diff
@@ -311,7 +311,7 @@ def test_enable_message_block(initialized_linter: PyLinter) -> None:
     # meth6
     assert not linter.is_message_enabled("E1101", 57)
     assert linter.is_message_enabled("E1101", 61)
-    assert not linter.is_message_enabled("E1101", 64)
+    assert linter.is_message_enabled("E1101", 64)
```
Comment (Pierre-Sassoulas) [inline tests/lint/unittest_lint.py:314]: Breaking change to warn about in changelog ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Looks great ! Wondering about the need for a changelog entry (not to warn about our internals change) but about the mc cabe and disable change in behavior

File: tests/functional/a/async_functions.py (hunk 1 above)
Comment (jacobtylerwalls) [reply tests/functional/a/async_functions.py:42]: Functional tests expected `too-many-branches`.  The try of a try/finally was double-counted before, so here adding an except back corrects the count and keeps the total identical.

File: tests/lint/unittest_lint.py (hunk 2 above)
Comment (jacobtylerwalls) [reply tests/lint/unittest_lint.py:314]: [a95880d](https://github.com/pylint-dev/pylint/pull/7767/commits/a95880d6d33b7adea0834b563f5298bf5c779bdf)

**Review by Pierre-Sassoulas — APPROVED** [review]
Great refactor, glad it's being merged after so much time/work !

**Pull request merged**